### PR TITLE
Make full model before calling set_model on callback

### PR DESCRIPTION
**Pull request by smatzek**
Commit 1b67ccbe8006eacffd268553abd01310e8b187d6 removed the _make_train_function calls from Keras training fit_generator for eager execution.

This breaks some callbacks that depend on the entire model to be populated on the set_model or on_train_begin methods.

This commit adds the method calls back in but guarded by an eager check.  It is not doing a revert / fix because the fix that removed the calls also put a test case in for eager fit_generator testing which we want to retain.

### CODE REVIEW

**Comment by wdirons** [comment]
@martinwicke , would you be able to review this change? We would like to get this cherry-picked into 1.10 if possible.

**Review by tanzhenyu — CHANGES_REQUESTED** [review]

File: tensorflow/python/keras/engine/training_generator.py [hunk 1]
```diff
@@ -49,6 +50,10 @@ def fit_generator(model,
   epoch = initial_epoch
 
   do_validation = bool(validation_data)
+  if not context.executing_eagerly():
```
[the remark this answers is not in the document]
Comment (smatzek) [reply tensorflow/python/keras/engine/training_generator.py]: Yes I can write a test for this.

[the remark this answers is not in the document]
Comment (smatzek) [reply tensorflow/python/keras/engine/training_generator.py]: Done

**Comment by case540** [comment]
Pinging for review

**Comment by smatzek** [comment]
@ tanzhenyu I have completed the changes you requested.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @tanzhenyu: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/python/keras/engine/training_test.py [hunk 2]
```diff
@@ -1060,6 +1061,37 @@ def custom_generator():
                                  use_multiprocessing=False,
                                  workers=0)
 
+  def test_fit_generator_with_callback(self):
```
[the remark this answers is not in the document]
Comment (smatzek) [reply tensorflow/python/keras/engine/training_test.py]: Done

**Review by alextp — DISMISSED** [review]

**Review by alextp — APPROVED** [review]

**Review by fchollet — APPROVED** [review]
LGTM, thanks for the PR.
